### PR TITLE
show how to log in Program.cs /3

### LINKED ISSUE #26143: Show how to log in Program.cs
This is a [frequent request at StackOverflow](https://stackoverflow.com/questions/45220938/logging-within-program-cs).

How about the following code:

```
var builder = WebApplication.CreateBuilder(args);

var logger = LoggerFactory.Create(config =>
{
    config.AddConsole();
    config.AddConfiguration(builder.Configuration.GetSection("Logging"));
}).CreateLogger("Program");

var app = builder.Build();

app.MapGet("/", () => "Hello World!");

app.MapGet("/Test", async context =>
{
    logger.LogInformation("Testing logging in Program.c");
    await context.Response.WriteAsync("Testing");
});

app.Run();
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7184a308-a254-9350-a5cd-5bced1f369ae
* Version Independent ID: 726e3bf1-f367-d733-8933-bccc04da0e16
* Content: [Logging in .NET Core and ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-6.0)
* Content Source: [aspnetcore/fundamentals/logging/index.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/logging/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #26143
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-6.0&branch=pr-en-us-26149#log-in-programcs)

### CODE REVIEW

File: aspnetcore/fundamentals/logging/index.md [hunk 1]
```diff
@@ -125,6 +125,16 @@ In the preceding sample:
   * `AzureAppServicesBlob`
   * `ApplicationInsights`
 
+## Log in Program.cs
+
+The following example calls <xref:Microsoft.Extensions.Logging.ConsoleLoggerExtensions.AddConsole(ILoggerFactory)> in `Program.c` and logs the `/Test` endpoint:
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/logging/index.md]: @guardrex  Line 130: [Warning] Cross reference not found: 'Microsoft.Extensions.Logging.ConsoleLoggerExtensions.AddConsole(ILoggerFactory)'.

**Comment by Rick-Anderson** [comment]
@Tratcher @tarekgh approved this code in https://github.com/dotnet/AspNetCore.Docs/issues/26143 but you might want to take a quick look to see if that's how ASP.NET Core folks should be logging in Program.cs

File: aspnetcore/fundamentals/logging/index.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/fundamentals/logging/index.md]: That tool calculates/guesses some of them. I think it should be ...

```
<xref:Microsoft.Extensions.Logging.ConsoleLoggerExtensions.AddConsole%2A>
```

File: aspnetcore/fundamentals/logging/index.md [hunk 2]
```diff
@@ -125,6 +125,16 @@ In the preceding sample:
   * `AzureAppServicesBlob`
   * `ApplicationInsights`
 
+## Log in Program.cs
+
+The following example calls [AddConsole](/dotnet/api/microsoft.extensions.logging.consoleloggerextensions.addconsole) <!--<xref:Microsoft.Extensions.Logging.ConsoleLoggerExtensions.AddConsole(ILoggerFactory)>--> in `Program.c` and logs the `/Test` endpoint:
```
Comment (guardrex) [inline aspnetcore/fundamentals/logging/index.md:130]:
```suggestion
The following example calls <xref:Microsoft.Extensions.Logging.ConsoleLoggerExtensions.AddConsole%2A> in `Program.cs` and logs the `/Test` endpoint:
```

File: aspnetcore/fundamentals/logging/index.md [hunk 3]
```diff
@@ -125,6 +125,16 @@ In the preceding sample:
...
+
+[!code-csharp[](index/samples/6.x/WebLog/Program.cs?name=snippet1)]
+
+The following example calls [AddSimpleConsole](/dotnet/api/microsoft.extensions.logging.consoleloggerextensions.addsimpleconsole) <!--<xref:Microsoft.Extensions.Logging.ConsoleLoggerExtensions.AddSimpleConsole(ILoggingBuilder)>--> in `Program.c`, disables color output, and logs the `/Test` endpoint:
```
Comment (guardrex) [inline aspnetcore/fundamentals/logging/index.md:134]:
```suggestion
The following example calls <xref:Microsoft.Extensions.Logging.ConsoleLoggerExtensions.AddSimpleConsole%2A> in `Program.cs`, disables color output, and logs the `/Test` endpoint:
```

File: aspnetcore/fundamentals/logging/index/samples/6.x/WebLog/Program.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    config.AddConsole();
+}).CreateLogger("Program");
+
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
+
+app.MapGet("/Test", async context =>
+{
+    logger.LogInformation("Testing logging in Program.c");
```
Comment (guardrex) [inline aspnetcore/fundamentals/logging/index/samples/6.x/WebLog/Program.cs:18]:
```suggestion
    logger.LogInformation("Testing logging in Program.cs");
```

File: aspnetcore/fundamentals/logging/index/samples/6.x/WebLog/Program.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+var logger = loggerFactory.CreateLogger<Program>();
+
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
+
+app.MapGet("/Test", async context =>
+{
+    logger.LogInformation("Testing logging in Program.c");
```
Comment (guardrex) [inline aspnetcore/fundamentals/logging/index/samples/6.x/WebLog/Program.cs:43]:
```suggestion
    logger.LogInformation("Testing logging in Program.cs");
```

File: aspnetcore/fundamentals/logging/index/samples/6.x/WebLog/appsettings.Development.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft.AspNetCore": "Warning"
+     // , "Microsoft.Hosting.Lifetime": "Warning"
```
Comment (guardrex) [inline aspnetcore/fundamentals/logging/index/samples/6.x/WebLog/appsettings.Development.json:6]: 😕 Per the spec, isn't this illegal? 👮‍♂️😄

File: aspnetcore/fundamentals/logging/index.md [hunk 7]
```diff
@@ -125,6 +125,16 @@ In the preceding sample:
   * `AzureAppServicesBlob`
   * `ApplicationInsights`
 
+## Log in Program.cs
```
Comment (guardrex) [inline aspnetcore/fundamentals/logging/index.md:128]:
```suggestion
## Log in `Program.cs`
```

File: aspnetcore/fundamentals/logging/index.md [hunk 8]
```diff
@@ -125,6 +125,16 @@ In the preceding sample:
   * `AzureAppServicesBlob`
   * `ApplicationInsights`
 
+## Log in `Program.cs`
```
Comment (Tratcher) [inline aspnetcore/fundamentals/logging/index.md:128]: There should also be a sample that covers using the WebApplicaiton's logger.
```
var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();
app.Logger.LogInformation("Adding Routes");
app.MapGet("/", () => "Hello World!");
app.Logger.LogInformation("Starting the app");
app.Run();
```

**Review by Tratcher — APPROVED** [review]

**Pull request merged**